WeatherApp/Views/WeatherDetailView.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  WeatherDetailView.swift
+//  WeatherApp
+//
+//  Created by Lucija Balja on 04/08/2020.
+//  Copyright © 2020 Lucija Balja. All rights reserved.
+//
+
+import UIKit
+
+class WeatherDetailView: UIView {
+    let dateLabel = UILabel()
+    let timeLabel = UILabel()
+    let weatherIcon = UIImageView()
+    let cityLabel = UILabel()
+    let weatherDescription = UILabel()
+    let currentTempLabel = UILabel()
+    let minTempLabel = UILabel()
+    let maxTempLabel = UILabel()
+    let humidityTitle = UILabel()
+    let humidityLabel = UILabel()
+    var weatherData: WeatherData? {
+        didSet {
+            setLabelsText()
+            setDateTime()
+        }
+    }
+    
+    init() {
+        super.init(frame: .zero)
+        setupUI()
+        setupConstraints()
+    }
+    
+    func setLabelsText() {
+        guard let weatherData = weatherData else { return }
+        weatherIcon.image = UIImage(systemName: weatherData.icon)
+        cityLabel.text = weatherData.city
+        weatherDescription.text = weatherData.description
+        currentTempLabel.text = "\(weatherData.parameters.currentTemperature)°C"
+        minTempLabel.text = "\(weatherData.parameters.minTemperature)° min "
+        maxTempLabel.text = "max \(weatherData.parameters.maxTemperature)°"
+        humidityTitle.text = "HUMIDITY"
+        humidityLabel.text = "\(weatherData.parameters.humidity)%"
+    }
+    
+    func setDateTime() {
+        let formatter = DateFormatter()
+        let currentDate = Date()
+        formatter.dateStyle = .full
+        dateLabel.text = formatter.string(from: currentDate)
+        formatter.dateStyle = .none
+        formatter.timeStyle = .short
+        timeLabel.text = formatter.string(from: currentDate)
+    }
+    
+    func setupUI() {
+        translatesAutoresizingMaskIntoConstraints = false
+        backgroundColor = .systemBlue
+        
+        UISetup.setLabel(dateLabel)
+        UISetup.setLabel(timeLabel, size: 30)
+        UISetup.setImageView(weatherIcon)
+        UISetup.setLabel(cityLabel, size: 40)
+        UISetup.setLabel(weatherDescription)
+        UISetup.setLabel(currentTempLabel, size: 60)
+        UISetup.setLabel(maxTempLabel, size: 30)
+        UISetup.setLabel(minTempLabel, size: 28)
+        UISetup.setLabel(humidityTitle)
+        UISetup.setLabel(humidityLabel, size: 25)
+        
+        cityLabel.numberOfLines = 0
+        weatherDescription.numberOfLines = 0
+        minTempLabel.textColor = .systemGray4
+        
+        addSubview(dateLabel)
+        addSubview(timeLabel)
+        addSubview(weatherIcon)
+        addSubview(cityLabel)
+        addSubview(weatherDescription)
+        addSubview(currentTempLabel)
+        addSubview(minTempLabel)
+        addSubview(maxTempLabel)
+        addSubview(humidityTitle)
+        addSubview(humidityLabel)
+    }
+    
+    func setupConstraints() {
+        dateLabel.topAnchor.constraint(equalTo: self.topAnchor, constant: 10).isActive = true
```

Comment:
You have a few repeating numbers here like margins and centerXAnchor, you should extract those to a constant. That way If you were to change margins, you only have to do it in one place.